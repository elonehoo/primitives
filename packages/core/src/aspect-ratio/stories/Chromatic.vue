<script setup lang="ts">
import { AspectRatio } from '../index.ts'
import Img from './Img.vue'
import './styles.css'
</script>

<template>
  <div>
    <h1>Default ratio</h1>
    <div :style="{ width: '300px' }">
      <AspectRatio class="aspectRation__root">
        <p>Default ratio (1/1)</p>
      </AspectRatio>
    </div>

    <h1>Custom ratios</h1>
    <div :style="{ display: 'flex', gap: '20px' }">
      <div :style="{ width: '200px' }">
        <AspectRatio :ratio="1 / 2">
          <Img />
        </AspectRatio>
      </div>
      <div :style="{ width: '200px' }">
        <AspectRatio>
          <Img />
        </AspectRatio>
      </div>
      <div :style="{ width: '200px' }">
        <AspectRatio :ratio="16 / 9">
          <Img />
        </AspectRatio>
      </div>
      <div :style="{ width: '200px' }">
        <AspectRatio :ratio="2 / 1">
          <Img />
        </AspectRatio>
      </div>
    </div>
  </div>
</template>
