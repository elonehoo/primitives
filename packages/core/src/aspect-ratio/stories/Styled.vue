<script setup lang="ts">
import { AspectRatio } from '../index.ts'
import './styles.css'
</script>

<template>
  <div :style="{ width: '500px' }">
    <AspectRatio class="aspectRation__root">
      <h1>Default ratio (1/1)</h1>
    </AspectRatio>
  </div>
</template>
