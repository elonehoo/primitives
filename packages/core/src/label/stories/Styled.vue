<script setup lang="ts">
import { Label } from '../index.ts'
import './styles.css'
</script>

<template>
  <div>
    <Label class="label_rootClass">Label</Label>
  </div>
</template>
