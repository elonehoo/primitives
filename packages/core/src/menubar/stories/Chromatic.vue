<script setup lang="ts">
import { shallowRef } from 'vue'
import { foodGroups } from '../../menu/stories/foods.ts'
import TickIcon from '../../menu/stories/TickIcon.vue'
import {
  MenubarArrow,
  MenubarCheckboxItem,
  MenubarContent,
  MenubarGroup,
  MenubarItem,
  MenubarItemIndicator,
  MenubarLabel,
  MenubarMenu,
  MenubarPortal,
  MenubarRadioGroup,
  MenubarRadioItem,
  MenubarRoot,
  MenubarSeparator,
  MenubarSub,
  MenubarSubContent,
  MenubarSubTrigger,
  MenubarTrigger,
} from '../index.ts'
import './styles.css'
import '../../menu/stories/styles.css'

const checkboxItems = [
  { label: 'Bold', state: shallowRef(false) },
  { label: 'Italic', state: shallowRef(true) },
  { label: 'Underline', state: shallowRef(false) },
  { label: 'Strikethrough', state: shallowRef(false), disabled: true },
]
const files = ['README.md', 'index.js', 'page.css']
const file = shallowRef(files[1]!)

function setFile(value: string) {
  file.value = value
}

function log(value: string) {
  // eslint-disable-next-line no-console
  console.log(value)
}
</script>

<template>
  <div>
    <div :style="{ padding: '200px', paddingTop: '50px', paddingBottom: '800px' }">
      <h1>Uncontrolled</h1>
      <h2>Closed</h2>
      <MenubarRoot class="menubar_rootClass">
        <MenubarMenu>
          <MenubarTrigger class="menubar_triggerClass">
            File
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass">
                New Tab
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Window
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Incognito Window
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />

              <MenubarItem class="menu_itemClass">
                Print…
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu>
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h2>Open</h2>
      <MenubarRoot default-value="file" class="menubar_rootClass">
        <MenubarMenu value="file">
          <MenubarTrigger class="menubar_triggerClass">
            File
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent
              class="menu_contentClass"
              :avoid-collisions="false"
              :side-offset="2"
              @focus-outside="(event) => event.preventDefault()"
            >
              <MenubarItem class="menu_itemClass">
                New Tab
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Window
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Incognito Window
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />

              <MenubarItem class="menu_itemClass">
                Print…
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h1 :style="{ marginTop: '180px' }">
        Controlled
      </h1>
      <h2>Closed</h2>
      <MenubarRoot value="" class="menubar_rootClass">
        <MenubarMenu value="file">
          <MenubarTrigger class="menubar_triggerClass">
            File
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass">
                New Tab
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Window
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Incognito Window
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />

              <MenubarItem class="menu_itemClass">
                Print…
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h2>Open</h2>
      <MenubarRoot value="file" class="menubar_rootClass">
        <MenubarMenu value="file">
          <MenubarTrigger class="menubar_triggerClass">
            File
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass">
                New Tab
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Window
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Incognito Window
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />

              <MenubarItem class="menu_itemClass">
                Print…
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h1 :style="{ marginTop: '200px' }">
        Submenus
      </h1>
      <MenubarRoot value="edit" class="menubar_rootClass">
        <MenubarMenu value="file">
          <MenubarTrigger class="menubar_triggerClass">
            File
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass">
                New Tab
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Window
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                New Incognito Window
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />

              <MenubarItem class="menu_itemClass">
                Print…
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarSub open>
                <MenubarSubTrigger class="menu_subTriggerClass">
                  Find <span>→</span>
                </MenubarSubTrigger>
                <MenubarPortal>
                  <MenubarSubContent
                    class="menu_contentClass"
                    :side-offset="10"
                    :align-offset="-6"
                    :avoid-collisions="false"
                  >
                    <MenubarItem class="menu_itemClass">
                      Search the web…
                    </MenubarItem>
                    <MenubarSeparator class="menu_separatorClass" />
                    <MenubarItem class="menu_itemClass">
                      Find…
                    </MenubarItem>
                    <MenubarItem class="menu_itemClass">
                      Find Next
                    </MenubarItem>
                    <MenubarItem class="menu_itemClass">
                      Find Previous
                    </MenubarItem>
                    <MenubarSub open>
                      <MenubarSubTrigger class="menu_subTriggerClass">
                        Advanced <span>→</span>
                      </MenubarSubTrigger>

                      <MenubarPortal>
                        <MenubarSubContent
                          class="menu_contentClass"
                          :side-offset="10"
                          :align-offset="-6"
                          :avoid-collisions="false"
                        >
                          <MenubarItem class="menu_itemClass">
                            Regex
                          </MenubarItem>
                          <MenubarItem class="menu_itemClass">
                            Replace
                          </MenubarItem>
                          <MenubarArrow />
                        </MenubarSubContent>
                      </MenubarPortal>
                    </MenubarSub>
                    <MenubarArrow />
                  </MenubarSubContent>
                </MenubarPortal>
              </MenubarSub>

              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h2 :style="{ marginTop: '250px' }">
        RTL
      </h2>
      <div dir="rtl">
        <MenubarRoot value="edit" class="menu_rootClass" dir="rtl">
          <MenubarMenu value="file">
            <MenubarTrigger class="menubar_triggerClass">
              File
            </MenubarTrigger>
            <MenubarPortal>
              <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
                <MenubarItem class="menu_itemClass">
                  New Tab
                </MenubarItem>
                <MenubarItem class="menu_itemClass">
                  New Window
                </MenubarItem>
                <MenubarItem class="menu_itemClass">
                  New Incognito Window
                </MenubarItem>
                <MenubarSeparator class="menu_separatorClass" />

                <MenubarItem class="menu_itemClass">
                  Print…
                </MenubarItem>
              </MenubarContent>
            </MenubarPortal>
          </MenubarMenu>

          <MenubarMenu value="edit">
            <MenubarTrigger class="menubar_triggerClass">
              Edit
            </MenubarTrigger>
            <MenubarPortal>
              <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
                <MenubarItem class="menu_itemClass" disabled>
                  Undo
                </MenubarItem>
                <MenubarItem class="menu_itemClass">
                  Redo
                </MenubarItem>
                <MenubarSeparator class="menu_separatorClass" />
                <MenubarSub open>
                  <MenubarSubTrigger class="menu_subTriggerClass">
                    Find <span>→</span>
                  </MenubarSubTrigger>
                  <MenubarPortal>
                    <MenubarSubContent
                      class="menu_contentClass"
                      :side-offset="10"
                      :align-offset="-6"
                      :avoid-collisions="false"
                    >
                      <MenubarItem class="menu_itemClass">
                        Search the web…
                      </MenubarItem>
                      <MenubarSeparator class="menu_separatorClass" />
                      <MenubarItem class="menu_itemClass">
                        Find…
                      </MenubarItem>
                      <MenubarItem class="menu_itemClass">
                        Find Next
                      </MenubarItem>
                      <MenubarItem class="menu_itemClass">
                        Find Previous
                      </MenubarItem>
                      <MenubarSub open>
                        <MenubarSubTrigger class="menu_subTriggerClass">
                          Advanced <span>→</span>
                        </MenubarSubTrigger>

                        <MenubarPortal>
                          <MenubarSubContent
                            class="menu_contentClass"
                            :side-offset="10"
                            :align-offset="-6"
                            :avoid-collisions="false"
                          >
                            <MenubarItem class="menu_itemClass">
                              Regex
                            </MenubarItem>
                            <MenubarItem class="menu_itemClass">
                              Replace
                            </MenubarItem>
                            <MenubarArrow />
                          </MenubarSubContent>
                        </MenubarPortal>
                      </MenubarSub>
                      <MenubarArrow />
                    </MenubarSubContent>
                  </MenubarPortal>
                </MenubarSub>

                <MenubarSeparator class="menu_separatorClass" />
                <MenubarItem class="menu_itemClass">
                  Cut
                </MenubarItem>
                <MenubarItem class="menu_itemClass">
                  Copy
                </MenubarItem>
                <MenubarItem class="menu_itemClass">
                  Paste
                </MenubarItem>
              </MenubarContent>
            </MenubarPortal>
          </MenubarMenu>
        </MenubarRoot>
      </div>

      <h2 :style="{ marginTop: '250px' }">
        With labels
      </h2>
      <MenubarRoot value="food" class="menubar_rootClass">
        <MenubarMenu value="food">
          <MenubarTrigger class="menubar_triggerClass">
            Food
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarGroup v-for="(foodGroup, index) in foodGroups" :key="index">
                <MenubarLabel v-if="foodGroup.label" :key="foodGroup.label" class="menu_labelClass">
                  {{ foodGroup.label }}
                </MenubarLabel>
                <MenubarItem
                  v-for="food in foodGroup.foods"
                  :key="food.value"
                  class="menu_itemClass"
                  :disabled="food.disabled"
                  @select="log(food.label)"
                >
                  {{ food.label }}
                </MenubarItem>
                <MenubarSeparator v-if="index < foodGroups.length - 1" class="menu_separatorClass" />
              </MenubarGroup>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>

      <h2 :style="{ marginTop: '600px' }">
        With checkbox and radio items
      </h2>
      <MenubarRoot value="items" class="menubar_rootClass">
        <MenubarMenu value="items">
          <MenubarTrigger class="menubar_triggerClass">
            Items
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" @select="log('show')">
                Show fonts
              </MenubarItem>
              <MenubarItem class="menu_itemClass" @select="log('bigger')">
                Bigger
              </MenubarItem>
              <MenubarItem class="menu_itemClass" @select="log('smaller')">
                Smaller
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarCheckboxItem
                v-for="checkboxItem in checkboxItems"
                :key="checkboxItem.label"
                class="menu_itemClass"
                :checked="checkboxItem.state.value"
                :disabled="checkboxItem.disabled"
                @update:checked="(v) => {
                  checkboxItem.state.value = v
                }"
              >
                {{ checkboxItem.label }}
                <MenubarItemIndicator>
                  <TickIcon />
                </MenubarItemIndicator>
              </MenubarCheckboxItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarRadioGroup :value="file" @update:value="setFile">
                <MenubarRadioItem
                  v-for="el in files"
                  :key="el"
                  class="menu_itemClass" :value="el"
                >
                  {{ el }}
                  <MenubarItemIndicator>
                    <TickIcon />
                  </MenubarItemIndicator>
                </MenubarRadioItem>
              </MenubarRadioGroup>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>

        <MenubarMenu value="edit">
          <MenubarTrigger class="menubar_triggerClass">
            Edit
          </MenubarTrigger>
          <MenubarPortal>
            <MenubarContent class="menu_contentClass" :avoid-collisions="false" :side-offset="2">
              <MenubarItem class="menu_itemClass" disabled>
                Undo
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Redo
              </MenubarItem>
              <MenubarSeparator class="menu_separatorClass" />
              <MenubarItem class="menu_itemClass">
                Cut
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Copy
              </MenubarItem>
              <MenubarItem class="menu_itemClass">
                Paste
              </MenubarItem>
            </MenubarContent>
          </MenubarPortal>
        </MenubarMenu>
      </MenubarRoot>
    </div>
  </div>
</template>
