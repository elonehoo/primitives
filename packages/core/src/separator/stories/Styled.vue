<script setup lang="ts">
import { Separator } from '../index.ts'
import './styles.css'
</script>

<template>
  <div>
    <h1>Horizontal</h1>
    <p>The following separator is horizontal and has semantic meaning.</p>
    <Separator class="separator_root" orientation="horizontal" />
    <p>
      The following separator is horizontal and is purely decorative. Assistive technology will
      ignore this element.
    </p>
    <Separator class="separator_root" orientation="horizontal" decorative />

    <h1>Vertical</h1>
    <div :style="{ display: 'flex', alignItems: 'center' }">
      <p>The following separator is vertical and has semantic meaning.</p>
      <Separator class="separator_root" orientation="vertical" />
      <p>
        The following separator is vertical and is purely decorative. Assistive technology will
        ignore this element.
      </p>
      <Separator class="separator_root" orientation="vertical" decorative />
    </div>
  </div>
</template>
