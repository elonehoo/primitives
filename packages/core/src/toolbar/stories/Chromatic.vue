<script setup lang="ts">
import ToolbarExample from './ToolbarExample.vue'
</script>

<template>
  <div :style="{ padding: 50 }">
    <h1>Example</h1>
    <ToolbarExample />
    <ToolbarExample orientation="vertical" />

    <h1>Direction</h1>
    <h2>Prop</h2>
    <ToolbarExample dir="rtl" />

    <h2>Inherited: WIP</h2>
    <!-- <DirectionProvider dir="rtl"> -->
    <ToolbarExample />
    <!-- </DirectionProvider> -->
  </div>
</template>
