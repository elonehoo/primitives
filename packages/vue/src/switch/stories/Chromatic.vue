<script setup lang="ts">
import { OkuSwitch, OkuSwitchThumb } from '@oku-ui/switch'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Off</h2>
  <OkuSwitch class="switch">
    <OkuSwitchThumb class="switch-thumb" />
  </OkuSwitch>

  <h2>On</h2>
  <OkuSwitch class="switch" default-checked>
    <OkuSwitchThumb class="switch-thumb" />
  </OkuSwitch>

  <h1>Controlled</h1>
  <h2>Off</h2>
  <OkuSwitch class="switch" :checked="false">
    <OkuSwitchThumb class="switch-thumb" />
  </OkuSwitch>

  <h2>On</h2>
  <OkuSwitch class="switch" checked>
    <OkuSwitchThumb class="switch-thumb" />
  </OkuSwitch>

  <h1>Disabled</h1>
  <OkuSwitch class="switch" disabled>
    <OkuSwitchThumb class="switch-thumb" />
  </OkuSwitch>

  <h1>State attributes</h1>
  <h2>Unchecked</h2>
  <OkuSwitch class="switch-attr-styles">
    <OkuSwitchThumb class="switch-attr-styles" />
  </OkuSwitch>

  <h2>Checked</h2>
  <OkuSwitch class="switch-attr-styles" default-checked>
    <OkuSwitchThumb class="switch-attr-styles" />
  </OkuSwitch>

  <h2>Disabled</h2>
  <OkuSwitch class="switch-attr-styles" default-checked disabled>
    <OkuSwitchThumb class="switch-attr-styles" />
  </OkuSwitch>
</template>
