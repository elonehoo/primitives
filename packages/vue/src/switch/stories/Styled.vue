<script setup lang="ts">
import { OkuSwitch, OkuSwitchThumb } from '@oku-ui/switch'
import Label from './Label.vue'
</script>

<template>
  <p>This switch is nested inside a label. The state is uncontrolled.</p>
  <Label>
    This is the label{{ ' ' }}
    <OkuSwitch class="switch">
      <OkuSwitchThumb class="switch-thumb" />
    </OkuSwitch>
  </Label>
</template>
