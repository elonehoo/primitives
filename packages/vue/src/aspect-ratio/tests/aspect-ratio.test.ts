import { enableAutoUnmount, mount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuAspectRatio } from '../'

import Styled from '../stories/Styled.vue'
import CustomRatios from '../stories/CustomRatios.vue'
import Chromatic from '../stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const RATIO = 1 / 2

describe('okuAspectRatio', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = mount({
      components: {
        OkuAspectRatio,
      },
      setup() {
        return {
          RATIO,
        }
      },
      template: `
        <div :style="{ width: '500px' }">
          <OkuAspectRatio :ratio="RATIO">
            <span>Hello</span>
          </OkuAspectRatio>
        </div>
      `,
    }, {
      attachTo: document.body,
    })
  })

  it('should render OkuAspectRatio correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })
})

describe('okuAspectRatio Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = mount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('customRatios', () => {
    let wrapper: VueWrapper<InstanceType<typeof CustomRatios>>

    beforeEach(async () => {
      wrapper = mount(CustomRatios, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = mount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
