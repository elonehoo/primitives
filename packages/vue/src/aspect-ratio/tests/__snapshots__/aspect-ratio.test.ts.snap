// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuAspectRatio > should render OkuAspectRatio correctly 1`] = `
"<div style="width: 500px;">
  <div style="position: relative; width: 100%; padding-bottom: 200%;" data-oku-aspect-ratio-wrapper="">
    <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><span>Hello</span></div>
  </div>
</div>"
`;

exports[`okuAspectRatio Stories > chromatic > should render correctly 1`] = `
"<h1>Default ratio</h1>
<div style="width: 300px;">
  <div style="position: relative; width: 100%; padding-bottom: 100%;" data-oku-aspect-ratio-wrapper="">
    <div class="aspect-ratio" style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;">
      <p>Default ratio (1/1)</p>
    </div>
  </div>
</div>
<h1>Custom ratios</h1>
<div style="display: flex; gap: 20px;">
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 200%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 100%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 56.25%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 50%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
</div>"
`;

exports[`okuAspectRatio Stories > customRatios > should render correctly 1`] = `
"<div style="display: flex; gap: 20px;">
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 200%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 100%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 56.25%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
  <div style="width: 200px;">
    <div style="position: relative; width: 100%; padding-bottom: 50%;" data-oku-aspect-ratio-wrapper="">
      <div style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;"><img src="https://images.unsplash.com/photo-1605030753481-bb38b08c384a?&amp;auto=format&amp;fit=crop&amp;w=400&amp;q=80" alt="A house in a forest" style="object-fit: cover; width: 100%; height: 100%;"></div>
    </div>
  </div>
</div>"
`;

exports[`okuAspectRatio Stories > styled > should render correctly 1`] = `
"<div style="width: 500px;">
  <div style="position: relative; width: 100%; padding-bottom: 100%;" data-oku-aspect-ratio-wrapper="">
    <div class="aspect-ratio" style="position: absolute; top: 0px; right: 0px; left: 0px; bottom: 0px;">
      <h1>Default ratio (1/1)</h1>
    </div>
  </div>
</div>"
`;
